cmd/batcher/main.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2019 kubeflow.org.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"os"
+	"strconv"
+	"github.com/kubeflow/kfserving/pkg/batcher"
+	"github.com/kubeflow/kfserving/pkg/batcher/controllers"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var (
+	port          = flag.String("port", "8082", "Logger port")
+	componentHost = flag.String("component-host", "127.0.0.1", "Component host")
+	componentPort = flag.String("component-port", "8080", "Component port")
+	maxBatchsize  = flag.String("max-batchsize", "32", "Max Batchsize")
+	maxLatency    = flag.String("max-latency", "1.0", "Max Latency")
+)
+
+func main() {
+	flag.Parse()
+
+	logf.SetLogger(logf.ZapLogger(false))
+	log := logf.Log.WithName("entrypoint")
+
+	maxBatchsizeInt, err := strconv.Atoi(*maxBatchsize)
+	if err != nil || maxBatchsizeInt <= 0 {
+		log.Info("Invalid max batchsize", "max-batchsize", *maxBatchsize)
+		os.Exit(-1)
+	}
+
+	maxLatencyFloat64, err := strconv.ParseFloat(*maxLatency, 64)
+	if err != nil || maxLatencyFloat64 <= 0.0 {
```

Comment:
max latency probably should not have a upper bound

Reply:
Agreed. This can vary a lot depending on use case. We have some huge models where inference time is 6s per request. I am ok with a 1s latency there for batching.

Reply:
That is a lower bound, not a upper bound.